### PR TITLE
Record Typefeedback on binops

### PR DESCRIPTION
**Pull request by o-**
based on top of #183

### CODE REVIEW

**Comment by o-** [comment]
(only c805491 is new)

File: rir/src/ir/RuntimeFeedback.h [hunk 1]
```diff
@@ -33,9 +35,43 @@ struct CallFeedback {
...
+struct TypeFeedback {
+    static constexpr unsigned MaxTypes = 3;
+    uint8_t numTypes;
+
+    std::array<RecordedType, MaxTypes> seen;
+
+    TypeFeedback() {}
+
+    void record(SEXP e) {
+        RecordedType type(e);
```
Comment (mhyee) [inline rir/src/ir/RuntimeFeedback.h:66]: OK, so I was wondering how this was working and if there was some weird C++ black magic... but turns out it isn't working -- the constructor isn't defined!

**Review by mhyee — COMMENTED** [review]
Looks good, other than the missing constructor.

Now to figure out how to use this type information...

File: rir/src/ir/RuntimeFeedback.h (hunk 1 above)
Comment (o-) [reply rir/src/ir/RuntimeFeedback.h:66]: haha, yeah, forgot to commit a file

**Comment by o-** [comment]
ah, i guess it is time for the weekend. I somehow managed to push 3 times without the constructor. Now i finally managed to, but managed to include the fixes to your comments on #183 . I'll be damned, but I am gonna merge it like that :)

**Pull request merged**